tests/Basics/SupervisedBufferTest.cpp

```diff
@@ -0,0 +1,63 @@
+#include "gtest/gtest.h"
+
+#include "Basics/ResourceUsage.h"
+#include "Basics/SupervisedBuffer.h"
+
+#include <velocypack/Builder.h>
+#include <velocypack/Value.h>
+#include <velocypack/Slice.h>
+
+using arangodb::basics::Exception;
+
+using arangodb::ResourceMonitor;
+
+using arangodb::velocypack::Builder;
+using arangodb::velocypack::SupervisedBuffer;
+using arangodb::velocypack::Value;
+
+TEST(SupervisedBufferTest, CapacityAccounting_GrowAndClear) {
+  ResourceMonitor monitor(nullptr);
+  std::size_t initial = monitor.current();
+
+  {
+    SupervisedBuffer supervisedBuffer(monitor);
+    Builder builder(supervisedBuffer);
+
+    builder.openArray();
+    for (int i = 0; i < 200; ++i) {
+      builder.add(Value("abcde"));
+    }
+    builder.close();
+
+    ASSERT_GT(monitor.current(), initial);
+
+    supervisedBuffer.clear();
+    ASSERT_EQ(monitor.current(), initial);
+  }
+
+  ASSERT_EQ(monitor.current(), initial);
+}
+
+TEST(SupervisedBufferTest, EnforcesLimit_OnGrowth) {
```

Comment:
Good test.

I think there are some more options in a builder that can increase the buffer.

I would like to ask for some more detailed test.

e.g. something like this:
Open an array
Add some elements, Assert memory is low.
Add some more elements until resize needs to happen, Assert memory is higher.

Also add asserts that current memory is always >= builder.size()
So we are sure we are counting correctly.


Furthremore please check the Builder API for the method that allows to recycle the Buffer.
Call it in the test and assert that the Memory stays high.